File: Numerical/FINAL/spline-linear.cpp
#include <bits/stdc++.h>

using namespace std;

double splineLinear(vector<double> x, vector<double> y, double find) {
    for (int i = 0; i < x.size() - 1; i++) {
        if (find >= x[i] && find <= x[i + 1]) {
            return y[i] + (y[i + 1] - y[i]) / (x[i + 1] - x[i]) * (find - x[i]);
        }
    }
    return 0;
}

int main() {
    vector<double> x = {2, 4, 6, 8, 10};
    vector<double> y = {9.5, 8, 10.5, 39.5, 72.5};
    double find = 4.5;

    cout << splineLinear(x, y, find) << endl;
    
    return 0;
}

File: Numerical/FINAL/spline-linear.py
def splineLinear(x, y, find):
    for i in range(len(x)-1):
        if(find >= x[i] and find <= x[i+1]):
            return y[i] + (y[i+1]-y[i])/(x[i+1]-x[i]) * (find-x[i])

x = [2, 4, 6, 8, 10]
y = [9.5, 8, 10.5, 39.5, 72.5]
find = 4.5
print(splineLinear(x, y, find))